src/wp-includes/user.php

```diff
@@ -682,14 +682,25 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 		return $pre;
 	}
 
-	$userlist = implode( ',', array_map( 'absint', $users ) );
-	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
+	// Cleanup the users array. Remove duplicates and sort for consistent ordering.
+	$users = array_unique( array_filter( array_map( 'absint', (array) $users ) ) );
+	sort( $users );
 
-	$result = $wpdb->get_results( "SELECT post_author, COUNT(*) FROM $wpdb->posts $where AND post_author IN ($userlist) GROUP BY post_author", ARRAY_N );
+	$userlist    = implode( '_', $users );
+	$cache_key   = "count_many_users_posts_{$post_type}_{$public_only}_{$userlist}_" . get_current_user_id();
+	$cache_group = $public_only ? 'count_many_users_posts_public' : 'count_many_users_posts';
```

Comment:
As you are using the public private status in the key, you can use a single group here. I think `user-queries` is probably the best match.